### PR TITLE
fix/v1.4 and v1.5 common revisions to v1.4

### PR DESCRIPTION
**Pull request by nokchalatte**
# Description (개요) 

Add fixes that should be applied to 1.4 and subsequent version(s) (1.5 currently):
- Add missing endpoint docs in engine version 1.4
- Translate tables docs from english to korean for ko version
- Fix incorrect parameter name in tables docs (schema_name -> schema)

## Type of change (작업사항)

Please delete options that are not relevant and check out the type of change.

- [x] REST API docs fix 

## Writing Convention

Make sure to check out the [Writing Convention](https://github.com/smartmind-team/tech-wiki/blob/main/project/thanosql/docs/writing_convention.md). 

## Review Process

Review should be conducted by at least two people.

### CODE REVIEW

File: docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_file.md [hunk 1]
```diff
@@ -42,15 +42,19 @@ designated folder.
...
-      -H 'Content-Type: multipart/form-data' \
-      -F 'file=@Data File Path;type=file_type/Data File Type'
+      -H 'Content-Type: application/json'
     ```
 
-If "db commit" is set to True and "table name" and "column name" are specified, the given file is sent to Workspace storage and the file path is inserted into a column of the specified table.
+
+## __`POST` /file/__
+
+In order to upload a file only, use the below methods to send a file to the Workspace storage.
```
Comment (stevensjcho) [inline docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_file.md]:
```suggestion
In order to upload a file, use the below methods to send a file to the Workspace storage.
```

File: docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_file.md [hunk 2]
```diff
@@ -80,20 +79,14 @@ If "db commit" is set to True and "table name" and "column name" are specified,
...
     ```
 
-!!! faq "__FAQ__"
-    - In order to use a file within the Jupyter workspace, you must put '/'home/jovyan' in front of the path. 
-
-
-## __`POST` /file/delete__
-
-In order to delete a file only, use the below methods to delete a file from the Workspace storage. 
+If "db commit" is set to True and "table name" and "column name" are specified, the given file is sent to Workspace storage and the file path is inserted into a column of the specified table.
```
Comment (stevensjcho) [inline docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_file.md]:
```suggestion
If "db_commit" is set to True and "table_name" and "column_name" are specified, the given file is sent to Workspace storage and the file path is inserted into the specified column of the specified table.
```

File: docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_file.md [hunk 3]
```diff
@@ -119,35 +116,36 @@ In order to delete a file only, use the below methods to delete a file from the
...
     ```
 
+!!! faq "__FAQ__"
+    - In order to use a file within the Jupyter workspace, you must put '/'home/jovyan' in front of the path.
 
-If "db_commit" is set to True and "table_name" and "column_name" are specified, the given file is deleted from Workspace storage, and all rows in the table with the same value as the specified file path in the specified column will be deleted.
+
+## __`DELETE` /file/__
+
+In order to delete a file only, use the below methods to delete a file from the Workspace storage.
```
Comment (stevensjcho) [inline docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_file.md]:
```suggestion
In order to delete a file from the Workspace storage, use the below methods.
```

File: docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_file.md [hunk 4]
```diff
@@ -156,17 +154,14 @@ If "db_commit" is set to True and "table_name" and "column_name" are specified,
...
       -H 'accept: application/json' \
       -H 'Authorization: Bearer Issued_API_TOKEN' \
       -H 'Content-Type: application/json'
     ```
 
-
-## __`POST` /file/list__
-
-A list of files and folders is returned from a specified file path. The file path can be expressed using a regular expression.
+If "db_commit" is set to True and "table_name" and "column_name" are specified, the given file is deleted from Workspace storage, and all rows in the table with the same value as the specified file path in the specified column will be deleted.
```
Comment (stevensjcho) [inline docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_file.md]:
```suggestion
If "db_commit" is set to True and "table_name" and "column_name" are specified, the given file is deleted from Workspace storage, and all rows in the table with the same value as the given file path in the specified column will be deleted.
```

File: docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md [hunk 5]
```diff
@@ -77,3 +77,45 @@ ThanoSQL 쿼리를 실행하고 쿼리 로그를 응답으로 받습니다.
 !!! warning "__Warning__"
     - "__CONVERT__"를 사용해 만들어진 컬럼의 값들은 base64로 인코딩됩니다. 바이트 형식의 값을 사용하려면 base64의 b64decode를 사용하여 디코딩해야 합니다.
 
+
+## **`GET` /query/log**
+
+이 메서드는 모든 쿼리 로그의 paginated 목록을 검색합니다.
```
Comment (stevensjcho) [inline docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md]:
```suggestion
쿼리 로그 목록을 가져옵니다.
```

File: docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md [hunk 6]
```diff
@@ -77,3 +77,45 @@ ThanoSQL 쿼리를 실행하고 쿼리 로그를 응답으로 받습니다.
...
+    ```shell
+      curl -X 'GET' \
+      'https://{your-engine-url}/api/v1/query/log/?search={search}&offset={offset}&limit={limit}' \
+      -H 'accept: application/json' \
+      -H 'Authorization: Bearer Issued_API_TOKEN'
+    ```
+
+### __Parameters__
+
+- `search`: 반환된 쿼리 로그에 포함되어야 하는 단어입니다.
```
Comment (stevensjcho) [inline docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md]:
```suggestion
- `search`: 로그의 쿼리 값 내에서 검색할 문자열.
```

File: docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md [hunk 7]
```diff
@@ -80,3 +80,46 @@ If the query statement yields results (statements such as SELECT, LIST), then th
...
+    ```shell
+      curl -X 'GET' \
+      'https://{your-engine-url}/api/v1/query/log/?search={search}&offset={offset}&limit={limit}' \
+      -H 'accept: application/json' \
+      -H 'Authorization: Bearer Issued_API_TOKEN'
+    ```
+
+### __Parameters__
+
+- `search`: Word(s) that the returned query logs should contain.
```
Comment (stevensjcho) [inline docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md]:
```suggestion
- `search`: The string to search for in the logs' queries
```

File: docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md [hunk 8]
```diff
@@ -77,3 +77,45 @@ ThanoSQL 쿼리를 실행하고 쿼리 로그를 응답으로 받습니다.
...
+- `offset`: pagination count이 시작되는 오프셋입니다(기본값은 0).
```
Comment (stevensjcho) [inline docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md]:
```suggestion
- `offset`: 쿼리 로그를 조회를 시작할 기준점 (기본값: 0).
```

File: docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md [hunk 9]
```diff
@@ -77,3 +77,45 @@ ThanoSQL 쿼리를 실행하고 쿼리 로그를 응답으로 받습니다.
...
+- `limit`: 오프셋에서 시작하여 검색할 최대 항목 수입니다(기본값은 100, 최대 100).
```
Comment (stevensjcho) [inline docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_query.md]:
```suggestion
- `limit`: offset에서 시작하여 조회 할 결과의 개수 (기본값: 100, 최대 100).
```

File: docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_schema.md [hunk 10]
```diff
@@ -0,0 +1,36 @@
+---
+title: Schema APIs
+---
+
+# **Schema APIs**
+
+
+## **`GET` /schema**
+
+데이터베이스에서 모든 스키마 목록을 가져오려면 아래 메서드를 사용하세요.
```
Comment (stevensjcho) [inline docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_schema.md]:
```suggestion
데이터베이스에서 모든 스키마 목록을 가져옵니다.
```

File: docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_tables.md [hunk 11]
```diff
@@ -4,36 +4,36 @@ title: Table APIs
...
+            - `name`: 기본키의 제약 조건 이름
+            - `columns`: 기본 키를 구성하는 컬럼
         - `foreign_keys`: 
-            - `name`: The constraint name of the foreign key
-            - `reference_schema`: The schema that contains the `reference_table`
-            - `reference_table`: The table that contains the `reference_column`
-            - `reference_column`: The column that the foreign key is referring to
-            - `column`: The foreign key column
+            - `name`: 외래키의 제약 조건 이름
+            - `reference_schema`: `reference_table` 포함하는 스키마
```
Comment (stevensjcho) [inline docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_tables.md]:
```suggestion
            - `reference_schema`: `reference_table`을 포함하는 스키마
```

File: docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_tables.md [hunk 12]
```diff
@@ -4,36 +4,36 @@ title: Table APIs
...
+            - `reference_table`: `reference_column` 포함하는 테이블
```
Comment (stevensjcho) [inline docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_tables.md]:
```suggestion
            - `reference_table`: `reference_column`을 포함하는 테이블
```

File: docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_tables.md [hunk 13]
```diff
@@ -4,36 +4,36 @@ title: Table APIs
...
+            - `reference_column`: 외래키가 참조하는 컬럼
+            - `column`: 외래키 컬럼
 
-    _The above components that have a * next to its name will not be included in the body for the POST table API. Instead, it will be specified in the URL as query parameters._
+    _이름 옆에 *가 붙은 위의 구성 요소는 `POST` Table API의 본문(Body)에 포함되지 않습니다. 대신 URL에 쿼리 매개변수로 지정됩니다._
```
Comment (stevensjcho) [inline docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_tables.md]:
```suggestion
    _이름 옆에 *가 붙은 위의 구성 요소는 `POST` Table API의 본문(Body)에서는 포함되지 않습니다. 대신 URL에 쿼리 매개변수로 지정됩니다._
```

File: docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_tables.md [hunk 14]
```diff
@@ -416,4 +416,86 @@ To delete a table, use the method below. If no `schema_name` query parameter is
...
       -H 'accept: application/json' \
       -H 'Authorization: Bearer Issued_API_TOKEN'
-    ```
\ No newline at end of file
+    ```
+
+
+## **`GET` /table/{table_name}/records**
+
+스키마에서 테이블의 paginated 레코드를 검색합니다.
```
Comment (stevensjcho) [inline docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_tables.md]:
```suggestion
스키마에서 테이블의 paginated 된 데이터를 가져옵니다.
```

File: docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_tables.md [hunk 15]
```diff
@@ -416,4 +416,86 @@ To delete a table, use the method below. If no `schema_name` query parameter is
...
+      -H 'Authorization: Bearer Issued_API_TOKEN'
+    ```
+
+
+### __Parameters__
+
+- `table_name`: 레코드를 검색할 테이블 이름입니다.
+- `schema`: 테이블을 검색할 스키마입니다(기본값은 'public').
+- `offset`: pagination count이 시작되는 오프셋입니다(기본값은 0).
+- `limit`: 오프셋에서 시작하여 검색할 최대 항목 수입니다(기본값은 100, 최대 100).
```
Comment (stevensjcho) [inline docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_tables.md]:
```suggestion
- `offset`: 쿼리 로그를 조회를 시작할 기준점 (기본값: 0).
- `limit`: offset에서 시작하여 조회 할 결과의 개수 (기본값: 100, 최대 100).
```

File: docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_tables.md [hunk 16]
```diff
@@ -416,4 +416,86 @@ To delete a table, use the method below. If no `schema_name` query parameter is
...
+
+
+## **`GET` /table/{table_name}/records/csv**
+
+스키마에 있는 테이블의 레코드를 CSV 파일로 검색합니다.
```
Comment (stevensjcho) [inline docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_tables.md]:
```suggestion
스키마에 있는 테이블의 레코드를 CSV 파일로 가져옵니다.
```

File: docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_file.md [hunk 17]
```diff
@@ -191,8 +191,8 @@ A list of files and folders is returned from a specified file path. The file pat
 === "cURL"
 
     ```shell
-    curl -X 'POST' \
-      'https://{your-engine-url}/api/v1/file/list/?file_path={File Path}' \
+    curl -X 'DELETE' \
+      'https://{your-engine-url}/api/v1/file/?db_commit=True&file_path=File Path&table_name=Table Name&column_name=Column Name' \
```
Comment (stevensjcho) [inline docs/en/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_file.md]: Note that the other curl examples use {} to show parameters

**Review by stevensjcho — APPROVED** [review]
LGTM 👍. We'll wait for @lazyduo 's review before merging to main.

File: docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_tables.md [hunk 18]
```diff
@@ -105,18 +105,18 @@ Use this method to get the objects of a single table. If no `schema_name` query
...
 
     DROP PRIMARY KEY -> DROP FOREIGN KEY -> DROP COLUMN -> ADD COLUMN -> ALTER COLUMN -> RENAME COLUMN -> ADD PRIMARY KEY -> ADD FOREIGN KEY -> RENAME TABLE -> SET SCHEMA
 
 
 !!! warning ""
-    When dealing with columns, it is important to note that the id is the unique key that is used to identify the column. If this value is deleted, that means that the column will also be deleted. If you are making a change to the column, **make sure that the column id is present in the body**! When adding a new column, a column_id is not required as it will be assigned after the column is created. 
+    컬럼을 다룰 때 `id`는 열을 식별하는 데 사용되는 고유 키라는 점에 유의하는 것이 중요합니다. 이 값이 삭제되면 열도 삭제된다는 의미입니다. 열을 변경하는 경우 **컬럼 `id`가 본문에 있는지 확인하세요**! 새 컬럼을 추가할 때는 컬럼이 생성된 후 할당되므로 컬럼 id는 필요하지 않습니다.
 
-In the following example lets pretend we want to alter the table object below:
+다음 예제에서는 아래의 테이블 객체를 변경한다고 가정해 보겠습니다:
```
Comment (lazyduo) [inline docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_tables.md]: There is absolutely no explanation of what kind of change is being made.
Right before the python/curl code, it is necessary to explain what kind of change it is.
(both en & ko)

**Review by lazyduo — COMMENTED** [review]
리뷰 및 수정 진행했습니다. 코멘트 내용만 추가하고 머지 진행하죠

File: docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_tables.md (hunk 18 above)
Comment (stevensjcho) [reply docs/ko/how-to_guides/ThanoSQL_connecting/rest_api_thanosql_tables.md]: 수정 했습니다, 내용 확인 부탁드려요

**Pull request merged**